### ISSUE TITLE
[DocDB] Alive Index can be treated as DELETED

### ISSUE THREAD

**Issue by OlegLoginov** [yugabyte/yugabyte-db#16163]
Jira Link: [DB-5599](https://yugabyte.atlassian.net/browse/DB-5599)
### Description

Sub-issue from: https://github.com/yugabyte/yugabyte-db/issues/13891

When calling `IsDeleteTableInProgress` on living table, it treats the table as being DELETED already.
```
Status CatalogManager::IsDeleteTableDone(...) {
...
 if (!l->started_deleting() && !l->started_hiding()) {    <<<
    Status s = STATUS(IllegalState, "The object was NOT deleted", l->pb.state_msg());
    return SetupError(resp->mutable_error(), MasterErrorPB::OBJECT_NOT_FOUND, s);   <<<
```
```
Status YBClient::Data::IsDeleteTableInProgress(...) {
...  auto status = SyncLeaderMasterRpc(
      deadline, req, &resp, "IsDeleteTableDone",
      &master::MasterDdlProxy::IsDeleteTableDoneAsync);
  if (resp.has_error()) {
    if (resp.error().code() == MasterErrorPB::OBJECT_NOT_FOUND) {    <<<
      *delete_in_progress = false;         <<< DELETED!
      return Status::OK();          <<< IsDeleteTableDone == DONE!
```

As a result, `WaitForDeleteTableToFinish` returns successfully and does so immediately.
This behavior is misleading & can hide a real issue.

A fix: calling of `IsDeleteTableInProgress` on a living table should error out.
With the error propagating out of `WaitForDeleteTableToFinish` as well.

Since previous OBJECT_NOT_FOUND master error made little sense, let's use another error (e.g. `INVALID_REQUEST`).

Reproducer:
- `ybd --java-test org.yb.cql.TestPartialIndex#testMoreThanOneIndex -n 100`
- `ybd --java-test org.yb.cql.TestPartialIndex#testPartialIndexSelectionPolicy -n 100`

[DB-5599]: https://yugabyte.atlassian.net/browse/DB-5599?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by OlegLoginov** [comment]
WIP diff: https://phabricator.dev.yugabyte.com/D22198
But the issue should be fixed first:   https://github.com/yugabyte/yugabyte-db/issues/15883

**Comment by OlegLoginov** [comment]
The issue is fixed by https://github.com/yugabyte/yugabyte-db/commit/1a4e857d9df8423c456dd31851c932e833decbfe
I'm keeping it open till backports to 2.14 & 2.16 landing.

**Comment by OlegLoginov** [comment]
Fixed - in the master branch & in 2.14 & in 2.16.
